taglib-core/src/main/java/com/reevoo/client/ReevooMarkClient.java

```diff
@@ -69,12 +80,13 @@ String obtainReevooMarkData(GetMethod request) throws IOException {
             content = null;
         }
 
-        if (status >= 500 && cachedResponse != null) {
-            Cache.put(cacheKey, new ReevooMarkRecord(cachedResponse, time_to_live));
-            return cachedResponse.value;
+        if (status >= 400) {
+            return cacheFailedRequest(cacheKey, cachedResponse, status).getValue();
         }
 
-        Cache.put(cacheKey, new ReevooMarkRecord(content, time_to_live, status, review_count));
+        int time_to_live = secondsToLive(request);
+        int review_count = extractReviewCountHeader(request);
+        Cache.put(cacheKey, ReevooMarkRecord.createRecord(content, time_to_live, status, review_count));
```

Comment:
I would put this in a method cacheSuccesfullRequest to make it in line with cacheFailedRequest